hpx/parallel/algorithms/exclusive_scan.hpp

```diff
@@ -146,16 +150,16 @@ namespace hpx { namespace parallel { inline namespace v1
                     std::forward<ExPolicy>(policy),
                     make_zip_iterator(first, dest), count, init,
                     // step 1 performs first part of scan algorithm
-                    [op](zip_iterator part_begin, std::size_t part_size) -> T
+                    [op, conv](zip_iterator part_begin, std::size_t part_size) -> T
                     {
-                        T part_init = get<0>(*part_begin++);
+                        T part_init = hpx::util::invoke(conv, get<0>(*part_begin++));
```

Comment:
> I looked at an example from algorithms/reduce.hpp L72, and even here there is no such check.

There it is not a problem as `part_size` would drop to zero, thus avoiding to dereference the iterator produced by `++part_begin` inside `accumulate_n`.